src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -91,6 +91,32 @@ pub fn update_lockfile(manifest_path: &Path,
                                                   Method::Everything,
                                                   Some(&previous_resolve),
                                                   Some(&to_avoid)));
+
+    // Summarize what is changing for the user.
+    let print_change = |status: &str, msg: String| {
+        opts.config.shell().status(status, msg)
+    };
+    for (removed, added) in compare_dependency_graphs(&previous_resolve, &resolve) {
+        if removed.len() == 1 && added.len() == 1 {
+            try!(print_change("Updating", format!("{} v{} -> v{}",
+                                                  removed[0].name(),
+                                                  removed[0].version(),
```

Comment:
Another part of package ids is the source it comes from (e.g. `foo` from crates.io is different than `foo` from github). In addition to indicating the source, updates of git repositories probably want to print out the SHA that was updated to instead of the revision as that's the "precise" notion there. You should be able to access this through the `precise` field of the `SourceId`